src/screen.rs

```diff
@@ -375,7 +375,11 @@ impl Screen {
         }
 
         // If overflowing, give the prompt its own line to improve the situation.
-        let first_line_prompt_space = layout.left_prompt_space;
```

Comment:
Some preliminary comments; I still need to finish reading the last two commits;
I don't expect many changes will be necessary.